### PR TITLE
docs: deprecate old alerts and dash/charts reports

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Deprecates the old alerts and reports feature

- Adds deprecation comments to the config keys
- Adds deprecation comments on the code modules
- Adds deprecation log warnings when the feature is enabled
- Adds a flash message on the MVC list for all the 3 views
 
Screenshot 2021-03-03 at 16 38 14

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by nytai — APPROVED** [review]

File: superset/config.py [hunk 1]
```diff
@@ -614,18 +615,8 @@ class CeleryConfig:  # pylint: disable=too-few-public-methods
     CELERY_ACKS_LATE = False
     CELERY_ANNOTATIONS = {
         "sql_lab.get_sql_results": {"rate_limit": "100/s"},
```
Comment (willbarrett) [inline superset/config.py:617]: I think these removals break SEMVER - we shouldn't change the defaults for the configuration between majors. Let's keep the schedules, but perhaps add a deprecation?

File: superset/config.py [hunk 2]
```diff
@@ -917,26 +919,29 @@ class CeleryConfig:  # pylint: disable=too-few-public-methods
...
 
 # The MAX duration (in seconds) a email schedule can run for before being killed
 # by celery.
+#
+# Warning: This config key is deprecated and will be removed in version 2.0.0"
 EMAIL_ASYNC_TIME_LIMIT_SEC = 300
 
-# Email report configuration
-# From address in emails
-EMAIL_REPORT_FROM_ADDRESS = "reports@superset.org"
```
Comment (willbarrett) [inline superset/config.py:928]: Same comment with these config removals.

**Review by willbarrett — APPROVED** [review]
LGTM after we update to conform to SEMVER

File: superset/config.py (hunk 2 above)
Comment (dpgaspar) [reply superset/config.py:928]: This one is used anywhere, safe to just remove

**Comment by willbarrett** [comment]
@dpgaspar looks like this just needs a lint fix.

**Pull request merged**